web/apps/serve.go

```diff
@@ -57,6 +59,27 @@ func onboarding(c echo.Context) bool {
 	return c.QueryParam("registerToken") != ""
 }
 
+// handleIntent will allow iframes from another app if the current app is
+// opened as an intent
+func handleIntent(c echo.Context, i *instance.Instance, slug, intentID string) {
+	intent := &intents.Intent{}
+	if err := couchdb.GetDoc(i, consts.Intents, intentID, intent); err != nil {
+		return
+	}
+	allowed := false
+	for _, service := range intent.Services {
+		if slug == service.Slug {
+			allowed = true
+		}
+	}
+	if !allowed {
+		return
+	}
+	from := i.SubDomain(intent.Client).Host
+	hdr := fmt.Sprintf("%s %s", middlewares.XFrameAllowFrom, from)
+	c.Response().Header().Set(echo.HeaderXFrameOptions, hdr)
+}
+
```

Comment:
Ok, thanks. My bad, I misunderstood what was the role of the `X-Frame-Options` header. I thought that it was a header for the main opener window. Actually it's not, it is just used for embedded ones. And you deal the CSP and url's siblings thanks to your `splitHost` method. Now I get the whole, thing, thanks !